api/metric/number.go

```diff
@@ -646,3 +646,15 @@ func (n *Number) compareWithZero(kind NumberKind) int {
 		return 0
 	}
 }
+
+// - sign change
+
+func ChangeSign(kind NumberKind, nn Number) Number {
```

Comment:
It could be changed to `func (n *Number) ChangeSign(kind NumberKind)`.  It would be confusing to take a pointer receiver and not modify the receiver, so there would be no return value.

However, every use of this function (and every test) was creating a new value, so changing it to modify its receiver would mean lots of change in the code for little value, so I thought it would be better to name this like a constructor, `NewNumberSignChange(Kind, Number) Number`. Does this help?